### PR TITLE
TokenExpiredException inherits from SignatureVerificationException + Refactor 

### PR DESCRIPTION
**Pull request by dstj**
... and Refactor SignatureVerificationException to use properties instead of the Data collection directly

### CODE REVIEW

File: src/JWT/JWT.cs [hunk 1]
```diff
@@ -239,10 +239,12 @@ public static void Verify(string payloadJson, string decodedCrypto, string decod
         {
             if (decodedCrypto != decodedSignature)
             {
-	            var signatureVerificationException = new SignatureVerificationException("Invalid signature");
-				signatureVerificationException.Data.Add("Expected", decodedCrypto);
-				signatureVerificationException.Data.Add("Received", decodedSignature);
-	            throw signatureVerificationException;
+               var signatureVerificationException = new SignatureVerificationException("Invalid signature")
```
Comment (abatishchev) [inline src/JWT/JWT.cs]: Can you please to make here the same change as in #63?

File: src/JWT/SignatureVerificationException.cs [hunk 2]
```diff
@@ -4,9 +4,34 @@ namespace JWT
...
         public SignatureVerificationException(string message)
             : base(message)
         {
         }
+
+        public string Expected
+        {
+            get
+            {
+               if (Data.Contains(ExpectedKey))
```
Comment (abatishchev) [inline src/JWT/SignatureVerificationException.cs]: And here, please.

File: src/JWT/JWT.cs (hunk 1 above)
Comment (dstj) [inline src/JWT/JWT.cs]: I'll rebase on PR #63 because I'll move the helper method in the `SignationVerificationException`...

Comment (dstj) [inline src/JWT/JWT.cs]: Done, I forced pushed my PR, we'll see how github likes it...

**Comment by abatishchev** [comment]
btw wouldn't it easier to have 1 PR for all these changes? and close the another one. Just thinking out loud.

File: src/JWT/JWT.cs (hunk 1 above)
Comment (abatishchev) [inline src/JWT/JWT.cs]: I think that one good enough, especially if we'll reside the helper method here. Pelsse go ahead and rebase, I'll merge

**Comment by dstj** [comment]
@abatishchev, I've rebased the PR. Yes it would have been easier (for me anyway) to do only one PR, but I wanted to keep the commit separated as they are not the same intention....

**Comment by abatishchev** [comment]
Cool! Would you mind please to format all documents, and then I'll merge right away.

**Comment by abatishchev** [comment]
@dstj please format and I'll merge and publish

**Comment by dstj** [comment]
@abatishchev Sorry about the delay, it is done. A few spaces were added...

**Comment by abatishchev** [comment]
Thanks again for your contributions! I need to publish this now

**Comment by abatishchev** [comment]
These change have been released as 1.4.1-beta (I'm bit scared to remove the per-release marker). Thanks again!

**Pull request merged**